scripts/populatejsonfiles.py

```diff
@@ -0,0 +1,110 @@
+"""
+Populate Json Files
+-------------------
+Gather urls from web reources and dump into appropriate json files.
+"""
+
+import json
+from enum import Enum
+from pathlib import Path
+from typing import List
+from dataclasses import dataclass
+
+import requests
+from requests.models import Response
+from requests import HTTPError
+from rdflib import Graph
+from rdflib.plugins.sparql.processor import SPARQLResult
+
+JSON_DIR = Path(__file__).parent.parent.absolute()
+
+
+class SPARQL_QUERIES(Enum):
+    """
+    Helper to holding constants that are SPARQL queries
+    """
+
+    IS_A_QUDT_UNIT = """
+    SELECT ?url {
+        ?url a <http://qudt.org/schema/qudt/Unit>
+    }
+    """
+
+    IS_A_QUDT_QUANTITY_KIND = """
+    SELECT ?url {
+        ?url a <http://qudt.org/schema/qudt/QuantityKind> 
+    }
+    """
+
+@dataclass
+class Resource:
+    """
+    A class to represent values that make up the resources we want to 
+    retrieve urls from and which json files we want to store them into.
+    """
+
+    url: str
+    json_path: Path
+    query: str
+
+
+RESOURCES = [
+    Resource(
+        url="http://qudt.org/2.1/vocab/quantitykind",
+        json_path=Path(JSON_DIR / "quantity-kinds.json"),
+        query=SPARQL_QUERIES.IS_A_QUDT_QUANTITY_KIND.value,
+    ),
+    Resource(
+        url="http://qudt.org/2.1/vocab/unit",
+        json_path=Path(JSON_DIR / "units.json"),
+        query=SPARQL_QUERIES.IS_A_QUDT_UNIT.value,
+    ),
+]
+
+
+def get(url: str) -> (Response):
+    """
+    Helper to get a http response representing defined list of resources
+    """
+
+    r: Response = requests.get(url)
+    if not r.ok:
+        raise HTTPError(f'Could not get "{url}" with status code {r.status_code}')
+    return r
+
+
+def update_json(json_path: Path, url_list: List[str]):
+    """
+    Function to update the enum urls in a json file with any urls from the input
+    list that are not already present.
+    """
+
+    with open(json_path, "r") as jsonFile:
+        data = json.load(jsonFile)
+
+        for url in url_list:
+            if url not in data["uris"]["enum"]:
```

Comment:
I think this is going to get quite large, can you convert it into a set before going through this for-loop and check whether url is in that? That should make it speedier as we build everything up.

I was going to suggest we delete values we don't find in the ttl but I guess we should always be adding, so I think you've made the right choice there.